test/e2e/prometheus_test.go

```diff
@@ -84,3 +85,98 @@ func TestPrometheusVolumeClaim(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
+	pods, err := f.KubeClient.CoreV1().Pods(f.Ns).List(metav1.ListOptions{FieldSelector: "status.phase=Running"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var alm = `affinity:
+	   podAntiAffinity:
+	      preferredDuringSchedulingIgnoredDuringExecution:
+	      - podAffinityTerm:
+	          labelSelector:
+	            matchExpressions:
+	            - key: alertmanager
+	              operator: In
+	              values:
+	              - main`
+
+	var k8s = `affinity:
+	   podAntiAffinity:
+	      preferredDuringSchedulingIgnoredDuringExecution:
+	      - podAffinityTerm:
+	          labelSelector:
+	            matchExpressions:
+	            - key: prometheus
+	              operator: In
+	              values:
+	              - k8s`
```

Comment:
let's remove those variables, as you can assert against invariants using strings directly.

Reply:
Let me not agree with you here. from qe point of view and using approach "from general to details" better to have in test case whole pod config overview.

Reply:
i disagree here, but the code is in a much better shape now, thanks for addressing all the nits minus this one ;-) ! :tada:

Reply:
thank you too, congrats ;-)!